core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -111,11 +121,17 @@ class MetadataCache(brokerId: Int) extends Logging {
       aliveNodes.get(brokerId).flatMap(_.get(listenerName))
     }
 
+  private def brokerAlive(brokerId: Int): Boolean =
+    inReadLock(partitionMetadataLock) {
```

Comment:
Why do we need this read lock?

Reply:
@ijuma Actually, this method isn't used, so removed.